package/src/cpumon.cpp

```diff
@@ -63,60 +63,57 @@ std::map<std::string, double> const cpumon::get_json_average_stats(
-                                   "cpu cores"};
-  unsigned int nCPU = 0, nSiblings = 0, nCores = 0;
+  // Metrics that'll be kept from the lscpu output
+  const std::vector<std::string> metrics{
+      "Model name", "Socket(s)", "Core(s) per socket", "Thread(s) per core"};
+
+  // Useful function to determine if a string is purely a number
+  auto isNumber = [](const std::string& s) {
+    return !s.empty() && std::find_if(s.begin(), s.end(), [](unsigned char c) {
+                           return !std::isdigit(c);
+                         }) == s.end();
+  };
+
+  // Loop over the output, parse the line, check the key and store the value if
+  // requested
+  std::string key{}, value{};
 
-  // Loop over the file
-  std::string line;
-  while (std::getline(cpuInfoFile, line)) {
+  for (const auto& line : cmd_result.second) {
+    // Continue on empty line
     if (line.empty()) continue;
+
+    // Tokenize by ":"
     size_t splitIdx = line.find(":");
-    std::string val;
-    if (splitIdx != std::string::npos) {
-      val = line.substr(splitIdx + 1);
-      if (val.empty()) continue;
-      for (const auto& metric : metrics) {
-        if (line.size() >= metric.size() &&
-            line.compare(0, metric.size(), metric) == 0) {
-          if (metric == "processor")
-            nCPU++;
-          else if (metric == "siblings") {
-            if (nSiblings == 0) nSiblings = std::stoi(val);
-          } else if (metric == "cpu cores") {
-            if (nCores == 0) nCores = std::stoi(val);
-          } else if (hw_json["HW"]["cpu"][metric].empty()) {
-            hw_json["HW"]["cpu"][metric] =
-                std::regex_replace(val, std::regex("^\\s+|\\s+$"), "");
-          }
-        }  // end of metric check
-      }    // end of populating metrics
-    }      // end of seperator check
-  }        // end of reading cpuInfoFile
-
-  // The clear assumption here is that there is a single type of processor
-  // installed nCPU = nSockets * nSiblings nSiblings = nCoresPerSocket *
-  // nThreadsPerCore
-  // Next 4 lines are a workaround for Raspberry Pi
-  if (!nSiblings) nSiblings = nCPU;
-  if (!nCores) nCores = nCPU;
-  unsigned int nSockets = nCPU / nSiblings;
-  unsigned int nThreads = nSiblings / nCores;
-  hw_json["HW"]["cpu"]["nCPU"] = nCPU;
-  hw_json["HW"]["cpu"]["nSockets"] = nSockets;
-  hw_json["HW"]["cpu"]["nCoresPerSocket"] = nCores;
-  hw_json["HW"]["cpu"]["nThreadsPerCore"] = nThreads;
-
-  // Close the file
-  cpuInfoFile.close();
+    if (splitIdx == std::string::npos) continue;
+
+    // Read "key":"value" pairs
+    key = line.substr(0, splitIdx);
+    value = line.substr(splitIdx + 1);
+    if (key.empty() || value.empty()) continue;
+    key = std::regex_replace(key, std::regex("^\\s+|\\s+$"), "");
+    value = std::regex_replace(value, std::regex("^\\s+|\\s+$"), "");
+
+    // Fill the JSON with the information
+    for (const auto& metric : metrics) {
```

Comment:
This is quadratic in complexity, but given that the output is small and the metrics are small in number I think we can get away with it (also, it's only called once)

Reply:
That was my line of thinking as well. However, given that we have only a handful of variables (conditions), I can just as well put in an if statement here. What do you think?

Reply:
I switched to using a simple `if` here.